src/main/java/de/hysky/skyblocker/skyblock/dungeon/puzzle/IceFill.java

```diff
@@ -0,0 +1,214 @@
+package de.hysky.skyblocker.skyblock.dungeon.puzzle;
+
+import com.google.common.primitives.Booleans;
+import com.mojang.brigadier.Command;
+import de.hysky.skyblocker.SkyblockerMod;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.debug.Debug;
+import de.hysky.skyblocker.skyblock.dungeon.secrets.DungeonManager;
+import de.hysky.skyblocker.skyblock.dungeon.secrets.Room;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.render.RenderHelper;
+import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.util.DyeColor;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+
+public class IceFill extends DungeonPuzzle {
+    public static final IceFill INSTANCE = new IceFill();
+    private static final float[] RED_COLOR_COMPONENTS = DyeColor.RED.getColorComponents();
+    private static final BlockPos BOARD_1_ORIGIN = new BlockPos(16, 70, 9);
+    private static final BlockPos BOARD_2_ORIGIN = new BlockPos(17, 71, 16);
+    private static final BlockPos BOARD_3_ORIGIN = new BlockPos(18, 72, 25);
```

Comment:
can be reduced to this:
```suggestion
    private static final BlockPos[] BOARD_ORIGINS = {
            new BlockPos(16, 70, 9),
            new BlockPos(17, 71, 16),
            new BlockPos(18, 72, 25)
    };
```